File: ModuleUser/src/main/java/cn/mycommons/moduleuser/UserModuleLifeCycle.java
package cn.mycommons.moduleuser;

import android.app.Application;
import android.support.annotation.NonNull;

import cn.mycommons.modulebase.IModuleConfig;
import cn.mycommons.modulebase.simple.SimpleModuleLifeCycle;
import cn.mycommons.moduleservice.IUserService;

/**
 * UserModuleLifeCycle <br/>
 * Created by xiaqiulei on 2017-05-14.
 */
public class UserModuleLifeCycle extends SimpleModuleLifeCycle {

    public UserModuleLifeCycle(@NonNull Application application) {
        super(application);
    }

    @Override
    public void onCreate(@NonNull IModuleConfig config) {
        config.registerService(IUserService.class, UserServiceImpl.class);

        config.registerRouter("app://user", UserActivity.class);
    }
}

File: ModuleUser/src/main/java/cn/mycommons/moduleuser/UserModuleLifeCycle.kt
package cn.mycommons.moduleuser

import android.app.Application

import cn.mycommons.modulebase.IModuleConfig
import cn.mycommons.modulebase.simple.SimpleModuleLifeCycle
import cn.mycommons.moduleservice.IUserService

/**
 * UserModuleLifeCycle <br></br>
 * Created by xiaqiulei on 2017-05-14.
 */
class UserModuleLifeCycle(application: Application) : SimpleModuleLifeCycle(application) {

    override fun onCreate(config: IModuleConfig) {
        config.registerService(IUserService::class.java, UserServiceImpl::class.java)

        config.registerRouter("app://user", UserActivity::class.java)
    }
}